### ISSUE TITLE
Python 3.8 Support

### ISSUE THREAD

**Issue by thoward27** [readthedocs/readthedocs.org#6324]
## Details

Just trying to keep track of when you plan to support python 3.8. I really want to use the walrus operator in my code, but that currently breaks RTD.

Repository in question is https://github.com/thoward27/grade

## Expected Result

Python 3.8 to be available in the config file.

## Actual Result

Tis not available:
```
Error
Problem in your project's configuration. Invalid "python.version": expected one of (2, 2.7, 3, 3.5, 3.6, 3.7, pypy3.5), got 3.8
```

**Comment by stsewd** [comment]
We need to add the new version in our docker file https://github.com/readthedocs/readthedocs-docker-images/blob/master/Dockerfile and test that nothing breaks.

**Comment by thoward27** [comment]
If you have an ETA and are willing to share it, I would greatly appreciate it! 

Let me know if there's anything I can do to help move this along.

**Comment by humitos** [comment]
@thoward27 do you have a PR in your repo so I can use to test RTD with a 3.8 project?

**Comment by thoward27** [comment]
@humitos I do now!

**Comment by humitos** [comment]
@thoward27 In the next deploy, #6329 will be included and we will be able to test your project under our docker image 6.0rc1 which includes py3.8. Do you want to beta test it?

**Comment by thoward27** [comment]
Sure, I'd love to. How can I go about doing so?

**Comment by humitos** [comment]
@thoward27 Great! I will enable the new Docker image with Py3.8 support in your project once it's deployed and let you know.

**Comment by humitos** [comment]
Hi @thoward27! We just deployed this. What's your project's url on Read the Docs? I can set the docker image with py3.8 support in your project.

Once this is set in your project, you don't have to use the `python.version` in your YAML file.

**Comment by thoward27** [comment]
@humitos my url is: https://grade.readthedocs.io/en/latest/

Let me know once you've updated the docker image

**Comment by stsewd** [comment]
@thoward27 I've enabled the new image for your project. If you use a configuration file, you can use `3.8` or just `3` as the python option https://docs.readthedocs.io/en/stable/config-file/v2.html#python and it will use python 3.8 to build your project.

Build using 3.8
https://readthedocs.org/projects/grade/builds/9959367/

**Comment by mikeshardmind** [comment]
I'm curious if there's any more on this, or if we can opt our project into the same build as well in the meantime, we recently constrained to python3.8+ due to upstream issue in python which was fixed in 3.8

**Comment by thoward27** [comment]
I have not made any progress on my end. Currently, I am waiting on several other packages to get to 3.8 before I can test it. Can someone enable EXTERNAL_VERSION_BUILD for my account? As I understand it, that should try and build docs for PRs, which would let me test just RTD much faster.

**Comment by humitos** [comment]
@mikeshardmind please, let me know what's your project and I will enable the Beta Docker image that contains Python 3.8 in your project. Also, see https://github.com/readthedocs/readthedocs.org/issues/6324#issuecomment-553669309 for its configuration.

**Comment by humitos** [comment]
@thoward27 let me know what's  your project on Read the Docs and I'll enable it.

**Comment by mikeshardmind** [comment]
@humitos thanks for the response, RTD project link https://red-discordbot.readthedocs.io/en/latest/

**Comment by humitos** [comment]
@mikeshardmind done! please try it now and let me know if you were able to build with Py3.8

**Comment by mikeshardmind** [comment]
@humitos Thank you, the beta image appears to work as intended for us ([successful build](https://readthedocs.org/projects/red-discordbot/builds/10209276/)) and the docs are rendering as intended.

**Comment by thoward27** [comment]
@humitos https://github.com/thoward27/grade

**Comment by stsewd** [comment]
Just for clarification, we still are in beta because people not using a configuration file or using a configuration file with `python: 3` will get python 3.8 instead of python 3.7. That could break some builds (bc of something like using async as identifier). Probably we want to communicate the update to users before making the change.

**Comment by humitos** [comment]
> @humitos [thoward27/grade](https://github.com/thoward27/grade)

@thoward27 I need your project on Read the Docs.

**Comment by protolambda** [comment]
Hey all, I would like to build my new project using python 3.8, as it extensively uses the newer typing features, and it's simply not 3.7 compatible. Can the beta be enabled for my project?

Project link: https://github.com/protolambda/remerkleable
Read the docs: https://readthedocs.org/projects/remerkleable/

Thanks!

**Comment by VilhelmPrytz** [comment]
> Hey all, I would like to build my new project using python 3.8, as it extensively uses the newer typing features, and it's simply not 3.7 compatible. Can the beta be enabled for my project?
> 
> Project link: https://github.com/protolambda/remerkleable
> Read the docs: https://readthedocs.org/projects/remerkleable/
> 
> Thanks!

Does your documentation not support Python 3.7?

**Comment by humitos** [comment]
@protolambda done! see the output of a build with py3.8 at https://readthedocs.org/projects/remerkleable/builds/10290130/

**Comment by protolambda** [comment]
@VilhelmPrytz Not exactly sure, but it failed on 3.7, since I wanted to install doc requirements through the setup.py, which requires 3.8. And then there also is autodoc with 3.8 types, and just the inconvenience about being limited to different python versions side by side.

@humitos Awesome, thank you :)

**Comment by sayanarijit** [comment]
@humitos Hi... can you please enable the same for https://readthedocs.org/projects/viper-infrastructure-commander/ ?

**Comment by jmsmkn** [comment]
@humitos Would you be able to enable it for https://grand-challengeorg.readthedocs.io/en/latest/ ?

**Comment by stsewd** [comment]
Sorry for the late reply. Both projects are running with python 3.8 now

https://readthedocs.org/projects/viper-infrastructure-commander/builds/10357201/
https://readthedocs.org/projects/grand-challengeorg/builds/10357209/

**Comment by DevilXD** [comment]
Could I get Python 3.8 enabled for this one too?
https://readthedocs.org/projects/arez/

**Comment by Paebbels** [comment]
My project pyVHDLParser is failing to build, due to Python 3.7 instead of Python 3.8:

```python
 File "/home/docs/checkouts/readthedocs.org/user_builds/pyvhdlparser/checkouts/latest/pyVHDLParser/Blocks/__init__.py", line 39, in <module>
    from pyVHDLParser.Token             import CharacterToken, Token, SpaceToken, IndentationToken, LinebreakToken, CommentToken, TokenIterator
  File "/home/docs/checkouts/readthedocs.org/user_builds/pyvhdlparser/checkouts/latest/pyVHDLParser/Token/__init__.py", line 56
    while char := next(charIterator):
                ^
SyntaxError: invalid syntax
```

This is a new project under development, so I started to use latest language features, because it will be medium outdated when my project gets to a stable release.

Please enable Python 3.8 for my project too.

If it's not so much effort, you can enable it for all my projects listed at ReadTheDocs :).

**Comment by stsewd** [comment]
@Paebbels which are the URLs of your projects or your rtd user?

**Comment by Paebbels** [comment]
@stsewd my user at ReadTheDocs is also called `Paebbels` and my projects are:

Projects with active Python 3.8 usage:
- pyAttributes
- pyVHDLParser

Projects going to use Python 3.8:
- pyTerminalUI
- pyTokenizer
- pyMetaClasses
- pyGenericPath
- pyHTTPInterface
- pyHTTPServer
- pyHTTPRequestRouter
- Semaphore-Microservice
- pyIPCMI

Thanks a lot in advance :).

**Comment by achimnol** [comment]
Could you please enable Python 3.8 for my projects as well? (username / slug ID)

 * joongi / aiotools
 * joongi / backendai-client-sdk-for-python
 * joongi / backendai-client-sdk-for-python-ko
 * lablup-devops / sorna

**Comment by cjolowicz** [comment]
Would it be possible to open a separate ticket to track when Python 3.8 support is added? Subscribing here means that you get notified every time somebody requests enabling it for their specific project. Thank you :-)

**Comment by humitos** [comment]
> Would it be possible to open a separate ticket to track when Python 3.8 support is added?

Yes. Good idea.

**Please, if you need beta support for Python 3.8 contact us at support@readthedocs.org** indicating your username and the project you want to enable it.

**Comment by humitos** [comment]
@cjolowicz regarding official support for Py3.8, we need are blocked on #6551 that it's required for this first.

**Comment by humitos** [comment]
Hi all! We already have support for Python 3.8. You can use our `.readthedocs.yml` config file for this (see https://docs.readthedocs.io/en/stable/config-file/v2.html)

Summarizing, you need:

```yaml
# .readthedocs.yml
python:
  version: 3.8
```

This was implemented at https://github.com/readthedocs/readthedocs.org/pull/6654

**Comment by danjac** [comment]
My .readthedocs.yaml for django-turbo-response is:
```

python:
  version: 3.8
    
```

It's currently breaking with assignment (walrus operator) syntax. Was this actually enabled as mentioned above or do we still have to request 3.8+ support?
